code_generator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -241,6 +291,15 @@ namespace <#= qualifiedNamespace #>
 
                         entityManager.RemoveComponent<Authoritative<<#= componentDetails.TypeName #>>>(entity);
                         entityManager.AddComponentData(entity, new NotAuthoritative<<#= componentDetails.TypeName #>>());
+
+                        // Remove event senders
+<# foreach (var eventDetail in eventDetailsList) { #>
+                        {
+                            var eventSender = entityManager.GetComponentData<EventSender.<#= eventDetail.PascalCaseName #>>(entity);
+                            ReferenceTypeProviders.<#= eventDetail.PascalCaseName #>Provider.Free(eventSender.handle);                           
+                            entityManager.RemoveComponent<EventSender.<#= eventDetail.PascalCaseName #>>(entity);
+                        }
+<# } #>
```

Comment:
Do we want this to still happen if there is a weird authority state? 

Could we get into a state where we have an event sender but no authority tag component?

Reply:
That's fine if we expect that contract to be unbroken